payjoin-cli/src/main.rs

```diff
@@ -70,58 +70,81 @@ fn cli() -> ArgMatches {
                 .long("retry")
                 .short('e')
                 .action(clap::ArgAction::SetTrue)
                 .help("Retry the asynchronous payjoin request if it did not yet complete"),
         )
-        .subcommand(
-            Command::new("send")
-                .arg_required_else_help(true)
-                .arg(arg!(<BIP21> "The `bitcoin:...` payjoin uri to send to"))
-                .arg_required_else_help(true)
-                .arg(
-                    Arg::new("fee_rate")
-                        .long("fee-rate")
-                        .value_name("FEE_SAT_PER_VB")
-                        .help("Fee rate in sat/vB")
-                        .value_parser(value_parser!(f32)),
-                ),
-        )
-        .subcommand(
-            Command::new("receive")
-                .arg_required_else_help(true)
-                .arg(arg!(<AMOUNT> "The amount to receive in satoshis"))
-                .arg_required_else_help(true)
-                .arg(
-                    Arg::new("port")
-                        .long("port")
-                        .short('p')
-                        .num_args(1)
-                        .help("The local port to listen on"),
-                )
-                .arg(
-                    Arg::new("endpoint")
-                        .long("endpoint")
-                        .short('e')
-                        .num_args(1)
-                        .help("The `pj=` endpoint to receive the payjoin request")
-                        .value_parser(value_parser!(Url)),
-                ),
-        )
-        .get_matches()
+        .subcommand_required(true);
+
+    // Conditional arguments based on features
+    #[cfg(feature = "v2")]
+    {
+        cmd = cmd.arg(
+            Arg::new("ohttp_relay")
+                .long("ohttp-relay")
+                .help("The ohttp relay url")
+                .value_parser(value_parser!(Url)),
+        );
+    }
+
+    cmd = cmd.subcommand(
+        Command::new("send")
+            .arg_required_else_help(true)
+            .arg(arg!(<BIP21> "The `bitcoin:...` payjoin uri to send to"))
+            .arg_required_else_help(true)
+            .arg(
+                Arg::new("fee_rate")
+                    .long("fee-rate")
+                    .value_name("FEE_SAT_PER_VB")
+                    .help("Fee rate in sat/vB")
+                    .value_parser(value_parser!(f32)),
+            ),
+    );
+
+    let mut receive_cmd = Command::new("receive")
+        .arg_required_else_help(true)
+        .arg(arg!(<AMOUNT> "The amount to receive in satoshis"))
+        .arg_required_else_help(true);
+
+    // Conditional arguments based on features for the receive subcommand
+    #[cfg(not(feature = "v2"))]
+    {
+        receive_cmd = receive_cmd.arg(
+            Arg::new("port")
+                .long("port")
+                .short('p')
+                .num_args(1)
+                .help("The local port to listen on"),
```

Comment:
I don't think this is applicable to `pj_endpoint`, because that's meant to be the one that ends up in the bitcoin URI and will typically be 443 while port is the internal listener that'll probably be handled by a TLS terminating reverse proxy.